src/main/java/uk/gov/hmcts/reform/sscs/services/sftp/SftpChannelAdapter.java

```diff
@@ -82,48 +116,121 @@ public ChannelSftp getSftpChannel() {
+        return list(true, PROCESSED_DIR);
     }
 
     @SuppressWarnings("unchecked")
-    private List<Gaps2File> list(String path) {
-        ChannelSftp channel = getSftpChannel();
+    private List<Gaps2File> list(boolean closeSession, String path) {
+        ChannelSftp channel = null;
         List<ChannelSftp.LsEntry> ls;
         try {
+            channel = (closeSession) ? openConnectedChannel() : openConnectedChannel(CHANNEL_1);
             ls = channel.ls(String.format("%s*.xml", path));
-        } catch (SftpException e) {
+            return ls.stream().map(e -> new Gaps2File(e.getFilename(), e.getAttrs().getSize()))
+                    .sorted()
+                    .collect(Collectors.toList());
+        } catch (JSchException | SftpException e) {
             throw new SftpCustomException("Failed reading incoming directory", e);
+        } catch (Exception e) {
+            return null;
+        } finally {
+            if (closeSession) {
+                close();
+            }
         }
-        return ls.stream().map(e -> new Gaps2File(e.getFilename(), e.getAttrs().getSize()))
-            .sorted()
-            .collect(Collectors.toList());
+
     }
 
     public InputStream getInputStream(String fileName) {
-        ChannelSftp channel = getSftpChannel();
+        return getInputStream(fileName, true);
+    }
+
+    private InputStream getInputStream(String fileName, boolean closeSession) {
+        Session session = null;
+        ChannelSftp sftp = null;
         try {
-            return channel.get(fileName);
-        } catch (SftpException e) {
+            session = openSession();
+            sftp = (closeSession) ? openConnectedChannel() : openConnectedChannel(CHANNEL_IN);
+            return new CloseableInputStream(sftp.get(fileName), session, sftp, closeSession);
+        } catch (JSchException | SftpException e) {
             throw new SftpCustomException("Failed reading file stream", fileName, e);
         }
     }
 
     public void move(boolean success, String fileName) {
-        ChannelSftp sftpChannel = getSftpChannel();
+        move(success, fileName, true);
+    }
+
+    private void move(boolean success, String fileName, boolean closeSession) {
+        ChannelSftp sftp = null;
         try {
-            sftpChannel.put(DUMMY, String.format("%s%s", success ? PROCESSED_DIR : FAILED_DIR, fileName));
-        } catch (SftpException e) {
+            sftp = (closeSession) ? openConnectedChannel() : openConnectedChannel(CHANNEL_1);
+            sftp.put(DUMMY, String.format("%s%s", success ? PROCESSED_DIR : FAILED_DIR, fileName));
+        } catch (JSchException | SftpException e) {
             throw new SftpCustomException("Failed moving file", fileName, e);
+        } finally {
+            if (closeSession) {
+                close();
+            }
         }
     }
 
+    public void close() {
+        close(true);
+    }
+
+    private void close(boolean success) {
```

Comment:
it seems like this private method always passes true? if so I would suggest inlining the method to the close method to simplify the code